Lib/test/test_posix.py

```diff
@@ -1476,6 +1478,106 @@ def test_empty_file_actions(self):
         )
         self.assertEqual(os.waitpid(pid, 0), (pid, 0))
 
+    def test_resetids(self):
```

Comment:
I did not find any immediate way to test this because it involves starting the parent with a particular real user id and a different effective user id and check the child. If you suggest a strategy I can implement it.

Reply:
You could try changing effective user id to some common Unix name (maybe _nobody_?), which should be possible if the test process has permission (e.g. root user). Either do it in a subprocess, or change user id back again after the test, so that further tests are unaffected. It looks like there's still a buildbot that runs as root.

Reply:
This is possible but I personally don't like the necessity to run the tests as root as if something goes wrong for any reason damage can be done. Also, this test won't run on the majority of CI due to by default a lot of them do not run as root. Also, `test_os.py` is not testing `os.setuid` as well for similar reasons I suppose.